docker/Dockerfile.agent.multiarch.rootless

```diff
@@ -1,4 +1,8 @@
-FROM --platform=$BUILDPLATFORM docker.io/golang:1.23 AS build
+FROM --platform=$BUILDPLATFORM docker.io/golang:1.23 AS builder
```

Comment:
Why do you call it builder now while it was called build in most places already and and still is named that way in some places?